pyang/plugins/tree.py

```diff
@@ -403,6 +407,22 @@ def get_typename(s):
                 return "-> %s" % "/".join(target)
             else:
                 return t.arg
+        elif verbose_types and t.arg == 'enumeration':
+            ret = t.arg
+            es = t.search('enum')
+            if es is not None:
```

Comment:
search() returns a list of statements, and if there is no 'enum' statement this list will be empty.  So 'if es is not None' will always be true.

Reply:
Did you consider using 'width' in the formatting of the enumerations?  Maybe create the string and if it is too wide fallback to the old format?  Or split into multiple lines.